100_EDA_Cleaning/104-EDA.Rmd

```diff
@@ -0,0 +1,493 @@
+eda_race_level$Driver.of.The.Day[eda_race_level$constructorId.x != eda_race_level$constructorId.y]
+#TODO: dig into this more
+
+
+#Q: Why do we have NA's in this dataset?
+eda_race_level[is.na(eda_race_level$points),]
+#A: The NA's are due to when we pulled Kaggle vs the most recent driver of the day.
+```
+
+
+## Begin EDA
+
+### Driver Level Checks
+
+```{r}
+#reformat data to be factors
+summary(eda_driver_level)
+eda_driver_level$driverId <- as.factor(eda_driver_level$driverId)
+eda_driver_level$Season <- as.factor(eda_driver_level$Season)
+
+summary(eda_driver_level)
+
+#EDA: can we look at age at time of race? (are younger drivers fan favorites?)
+#hist(as.numeric(eda_race_level$age_at_race))
+
+#EDA: number of stops?
+hist(eda_race_level$num_stops)
+
+#EDA: time of race?
+table(eda_race_level$afternoon_race)
+
+#EDA: Finishing the race/status?
+table(eda_race_level$status)
+```
+
+###Driver of the Day Observations
+```{r}
+glimpse(eda_race_level)
+sort(table(eda_race_level$Driver.of.The.Day), decreasing=TRUE)
+unique(eda_race_level$Season)
+#Driver of the day data is only available at the earliest 2016. All data sets will only use 2016 to the most recent.
+```
+
+Verstappen has many driver of the day awards in all years. The overall distribution of the winners changes per years. 
+```{r}
+#Distribution of Driver of the day by season and race
+ggplot(data = filter(eda_race_level), aes(x = Driver.of.The.Day)) + geom_bar() + 
+    facet_wrap(~Season) + xlab("Driver") + ylab("Count") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + ggtitle("Distribution of Race in regard of Driver of the Day in 2016-2022") +  theme(plot.title = element_text(hjust = 0.5)) 
+```
+
+Verstappen received driver of the day the most, followed by Vettel and Hamilton.
+```{r}
+#Count of driver of the day 
+driver_day <- eda_race_level %>% group_by(Driver.of.The.Day) %>% summarize(count_driver_day = n()) %>% arrange(desc(count_driver_day))
+ggplot(data = filter(driver_day), aes(x = reorder(Driver.of.The.Day,-count_driver_day),y=count_driver_day)) + geom_bar(stat='identity') + xlab("Driver") + ylab("Count") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + ggtitle("Driver of the Day in 2016-2022") + theme(plot.title = element_text(hjust = 0.5))
+```
+Drivers who were drivers of the day the most by year:
+2016: Verstappen
+2017: Vettel
+2018: Verstappen
+2019: Verstappen
+2020: Verstappen
+2021: Perez
+2022: Leclerc and Verstappen
+
+```{r}
+driver_day2 <- eda_race_level %>% group_by(Season,Driver.of.The.Day) %>% summarize(count_driver_day = n()) %>% arrange(desc(count_driver_day))
+ggplot(data = filter(driver_day2), aes(x = as.factor(Season),y=count_driver_day,fill=Driver.of.The.Day)) + geom_bar(stat='identity') + xlab("Season") + ylab("Count") + scale_x_discrete(guide = guide_axis(n.dodge=2)) + ggtitle("Distribution of Driver of the Day by Season") + theme(plot.title = element_text(hjust = 0.5))
```

Comment:
**Nit** For readability/style, consider having a newline between each section of the call

```
ggplot(data = filter(driver_day2), aes(x = as.factor(Season),y=count_driver_day,fill=Driver.of.The.Day)) +
       geom_bar(stat='identity') +
        xlab("Season") +
        ylab("Count") + 
       scale_x_discrete(guide = guide_axis(n.dodge=2)) +
       ggtitle("Distribution of Driver of the Day by Season") + 
       theme(plot.title = element_text(hjust = 0.5))
```